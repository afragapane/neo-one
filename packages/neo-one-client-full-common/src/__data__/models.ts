import { common, ContractParameterTypeModel, ECPoint, SignatureString, UInt160 } from '@neo-one/client-common';
import { constants } from '@neo-one/utils';
import {
  ContractABIModel,
  ContractEventModel,
  ContractGroupModel,
  ContractManifestModel,
  ContractMethodDescriptorModel,
  ContractModel,
  ContractParameterDeclarationModel,
  ContractPermissionDescriptorModel,
  ContractPermissionsModel,
  ContractPropertyStateModel,
} from '../models';

export const contractParamModel = {
  boolean: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Boolean,
    name: 'param',
  }),
  byteArray: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.ByteArray,
    name: 'param',
  }),
  hash160: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Hash160,
    name: 'param',
  }),
  hash256: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Hash256,
    name: 'param',
  }),
  array: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Array,
    name: 'param',
  }),
  integer: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Integer,
    name: 'param',
  }),
  interopInterface: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.InteropInterface,
    name: 'param',
  }),
  map: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Map,
    name: 'param',
  }),
  publicKey: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.PublicKey,
    name: 'param',
  }),
  signature: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Signature,
    name: 'param',
  }),
  string: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.String,
    name: 'param',
  }),
  void: new ContractParameterDeclarationModel({
    type: ContractParameterTypeModel.Void,
    name: 'param',
  }),
};

export const contractMethodDescriptorModel = (
  parameters: readonly ContractParameterDeclarationModel[] = [contractParamModel.boolean],
  returnType: ContractParameterTypeModel = ContractParameterTypeModel.Void,
  name = 'function',
) => new ContractMethodDescriptorModel({ name, parameters, returnType });

export const contractEventModel = (
  parameters: readonly ContractParameterDeclarationModel[] = [contractParamModel.boolean],
  name = 'event',
) => new ContractEventModel({ name, parameters });

export const contractAbiModel = (
  methods: readonly ContractMethodDescriptorModel[] = [contractMethodDescriptorModel()],
  events: readonly ContractEventModel[] = [contractEventModel()],
  entryPoint: ContractMethodDescriptorModel = contractMethodDescriptorModel(),
  hash = common.bufferToUInt160(Buffer.alloc(20, 1)),
) => new ContractABIModel({ hash, entryPoint, methods, events });

export const contractGroupModel = (
  publicKey: ECPoint = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY),
  signature: SignatureString = 'ccaab040cc25021c91567b75db4778853441869157b8f6aad960cdcf1069812480027a528ca9b98e2205027de20696f848cf81824eeb7af1d5110870870ceb67',
) => new ContractGroupModel({ publicKey, signature });

export const contractPermissionDescriptorModel = (hashOrGroupType: 'uint160' | 'ecpoint' | undefined) => {
  let hashOrGroup: UInt160 | ECPoint | undefined;
  if (hashOrGroupType === 'uint160') {
    hashOrGroup = common.bufferToUInt160(Buffer.alloc(20, 1));
  } else if (hashOrGroupType === 'ecpoint') {
    hashOrGroup = common.stringToECPoint(constants.PRIVATE_NET_PUBLIC_KEY);
  }

  return new ContractPermissionDescriptorModel({ hashOrGroup });
};

export const contractPermissionsModel = (
  hashOrGroupType: 'uint160' | 'ecpoint' | undefined,
  methods: readonly string[] = [],
) => new ContractPermissionsModel({ contract: contractPermissionDescriptorModel(hashOrGroupType), methods });

export const contractManifestModel = (
  groups: readonly ContractGroupModel[] = [contractGroupModel()],
  features: ContractPropertyStateModel = ContractPropertyStateModel.HasStoragePayable,
  abi: ContractABIModel = contractAbiModel(),
  permissions: readonly ContractPermissionsModel[] = [contractPermissionsModel('uint160', ['method1'])],
  trusts: readonly UInt160[] = [common.bufferToUInt160(Buffer.alloc(20, 1))],
  safeMethods: readonly string[] = ['method1', 'method2'],
) => new ContractManifestModel({ groups, features, abi, permissions, trusts, safeMethods });

export const contractModel = (
  script: Buffer = Buffer.alloc(25),
  manifest: ContractManifestModel = contractManifestModel(),
) => new ContractModel({ script, manifest });
