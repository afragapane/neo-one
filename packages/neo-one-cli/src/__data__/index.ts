export * from './getClients';
export * from './getContracts';
