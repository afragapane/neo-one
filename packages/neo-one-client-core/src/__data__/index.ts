export * from './keys';
export * from './data';
