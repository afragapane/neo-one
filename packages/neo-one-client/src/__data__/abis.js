/* @flow */
import type {
  ABIEvent,
  ABIFunction,
  ABIParameter,
  ABIReturn,
  ABI,
} from '../types';

export const returns = {
  Signature: { type: 'Signature' },
  Boolean: { type: 'Boolean' },
  Hash160: { type: 'Hash160' },
  Hash256: { type: 'Hash256' },
  ByteArray: { type: 'ByteArray' },
  PublicKey: { type: 'PublicKey' },
  String: { type: 'String' },
  InteropInterface: { type: 'InteropInterface' },
  Void: { type: 'Void' },
  Integer: { type: 'Integer', decimals: 7 },
  Array: { type: 'Array', value: { type: 'String' } },
};

export const parameters = {
  Signature: { name: 'signatureName', type: 'Signature' },
  Boolean: { name: 'booleanName', type: 'Boolean' },
  Integer: { name: 'intName', type: 'Integer', decimals: 7 },
  Hash160: { name: 'hash160Name', type: 'Hash160' },
  Hash256: { name: 'hash256Name', type: 'Hash256' },
  ByteArray: { name: 'byteName', type: 'ByteArray' },
  PublicKey: { name: 'pubkeyName', type: 'PublicKey' },
  String: { name: 'stringName', type: 'String' },
  Array: {
    name: 'arrayName',
    type: 'Array',
    value: {
      name: 'stringName',
      type: 'String',
    },
  },
  InteropInterface: { name: 'interopName', type: 'InteropInterface' },
  Void: { name: 'voidName', type: 'Void' },
};

export const abiFunction = (
  constant: boolean = false,
  params: Array<ABIParameter> = [parameters.String],
  verify: boolean = false,
  name: string = 'funcName',
  returnType: ABIReturn = returns.Void,
): ABIFunction => ({
  name,
  constant,
  parameters: params,
  verify,
  returnType,
});

export const abiEvent = (
  name: string = 'eventName',
  params: Array<ABIParameter> = [parameters.String],
): ABIEvent => ({
  name,
  parameters: params,
});

export const abi = (
  functions: Array<ABIFunction> = [abiFunction()],
  events?: Array<ABIEvent> = [],
): ABI => ({
  functions,
  events,
});
