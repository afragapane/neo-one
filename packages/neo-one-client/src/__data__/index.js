/* @flow */
export { default as keys } from './keys';
export { default as transactions } from './transactions';
