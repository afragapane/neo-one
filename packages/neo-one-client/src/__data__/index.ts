export * from './keys';
export * from './data';
export * from './transactions';
export * from './factory';
