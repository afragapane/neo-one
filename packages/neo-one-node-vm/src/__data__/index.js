/* @flow */
export { default as createBlockchain } from './createBlockchain';
export { default as keys } from './keys';
export * as transactions from './transactions';
export * as testUtils from './utils';
