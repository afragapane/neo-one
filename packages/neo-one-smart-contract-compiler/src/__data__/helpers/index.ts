export * from './compile';
export * from './createContext';
export * from './execute';
export * from './extractors';
export * from './getMonitor';
export * from './startNode';
export * from './transpileAndExecute';
export * from './utils';
