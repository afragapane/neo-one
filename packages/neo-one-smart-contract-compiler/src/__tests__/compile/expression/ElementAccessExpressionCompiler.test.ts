import { helpers } from '../../../__data__';

describe('ElementAccessExpressionCompiler', () => {
  test('[0, 1, 2][idx]', async () => {
    await helpers.executeString(`
      const x: Array<number> = [0] as Array<number>;

      assertEqual(x.length, 1);

      const y: Array<number> = x;

      y[2] = 2;
      if (y.length !== 3) {
        throw 'Failure';
      }

      if (y[0] !== 0) {
        throw 'Failure';
      }

      if (y[1] !== undefined) {
        throw 'Failure';
      }

      if (y[2] !== 2) {
        throw 'Failure';
      }

      if ((y[2] += 1) !== 3) {
        throw 'Failure';
      }

      if (y[2] !== 3) {
        throw 'Failure';
      }
    `);
  });

  test('{ a: 0, b: 1 }[element]', async () => {
    await helpers.executeString(`
      const x = { a: 0, b: 1 };
      if (x['a'] !== 0) {
        throw 'Failure';
      }

      if (x['b'] !== 1) {
        throw 'Failure';
      }

      if ((x['a'] += 1) !== 1) {
        throw 'Failure';
      }

      if (x['a'] !== 1) {
        throw 'Failure';
      }
    `);
  });

  test('{ [Symbol.for("a")]: 0 }[element]', async () => {
    await helpers.executeString(`
      const a = Symbol.for('a');
      const x = { [a]: 0 };
      if (x[a] !== 0) {
        throw 'Failure';
      }

      if ((x[a] += 1) !== 1) {
        throw 'Failure';
      }

      if (x[a] !== 1) {
        throw 'Failure';
      }
    `);
  });

  test('mixed index types', async () => {
    await helpers.executeString(`
      const a: string | number = 0 as string | number;
      const x = { [a]: 0 };
      if (x[a] !== 0) {
        throw 'Failure';
      }

      if ((x[a] += 1) !== 1) {
        throw 'Failure';
      }

      if (x[a] !== 1) {
        throw 'Failure';
      }

      x[a];
    `);
  });
});
