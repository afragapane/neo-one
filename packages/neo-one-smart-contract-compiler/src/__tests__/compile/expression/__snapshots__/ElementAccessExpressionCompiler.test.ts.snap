// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementAccessExpressionCompiler [0, 1, 2]["idx"] 1`] = `
"snippetCode.ts (3,9): Element implicitly has an 'any' type because index expression is not of type 'number'.

      1 | 
      2 |       const x = [0, 1, 2]
    > 3 |       x['0'];
        |         ^
      4 |       const length: string = 'length' as string;
      5 |       x[length];
      6 |     
"
`;

exports[`ElementAccessExpressionCompiler Object["keys"] 1`] = `
"snippetCode.ts (4,7): Cannot find name 'assertEqual'.

      2 |       const x = Object['keys'];
      3 | 
    > 4 |       assertEqual(x !== undefined, true);
        |       ^
      5 |     
"
`;

exports[`ElementAccessExpressionCompiler buffer["idx"] 1`] = `
"snippetCode.ts (3,7): Element implicitly has an 'any' type because type 'Buffer' has no index signature.

      1 | 
      2 |       const x = Buffer.from('3030', 'hex');
    > 3 |       x[1];
        |       ^
      4 |     
"
`;
